### ISSUE TITLE
Save images in Markdown

### ISSUE THREAD

**Issue by fischer-felix** [quillpad/quillpad#9]
Currently, whenever you add an image in the app, the note is just empty in Nextcloud.

It would be nice if any image from a quillnote would just be a Markdown image in the web interface.

The images in Nextcloud are stored in the user's home directory under `~/Notes/Notebook if applicable/` next to the actual Note.
I've had a look through the code and checked the notes API and it seems there is no way of uploading an image this way, so if I'm not wrong this means we would have to use a WebDAV connection.

**Comment by jsixface** [comment]
There is currently an issue in [Notes](/nextcloud/notes) to open up the API for attachments in notes. https://github.com/nextcloud/notes/issues/826 
Once that is available, it'll be easy to integrate that API. I don't think going the WebDAV route is the right option - that ties the app closely with Nextcloud architecture and I'd like to keep it generic.

**Comment by fischer-felix** [comment]
Absolutely, if there is already work on this in notes I'd just wait until this is available

**Comment by mohammadrafigh** [comment]
To stay generic the app can take a copy of attached files and images somewhere in a private directory so removing the image from gallery wouldn't break notes. This way syncing could easily happen by syncing the copied files directory as well. I see the syncing milestone is set to store markdown files in a directory and sync with other cloud drives. If it's going to be done then syncing attached images would be easily possible too. In current implementation it's not a good idea that I should take care of files and their paths to prevent my notes from being broken.

**Comment by serhii-ilin** [comment]
Looks like the corresponding Nextcloud Notes API issue is closed 
I wonder if there are any plans to implement this for Quilpad?
I don't have experience with Android, but I could try to do it if someone guides or gives pointers.
Thanks
